ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp

```diff
@@ -209,6 +222,61 @@ Tensor all_gather(
     return output_tensors.at(0);
 }
 
+Tensor all_gather(
+    const Tensor& input_tensor,
+    const uint32_t dim,
+    const uint32_t cluster_axis,
+    const MeshDevice& mesh_device,
+    const uint32_t num_links,
+    const std::optional<MemoryConfig>& memory_config,
+    const std::optional<size_t> user_defined_num_workers,
+    const std::optional<size_t> user_defined_num_buffers_per_channel,
+    const ttnn::ccl::Topology topology) {
+
+    TT_FATAL(topology == ttnn::ccl::Topology::Linear, "This api currently supported only for Linear topology");
```

Comment:
Please add more details to the error message. What API is this? The user may get confused